### PR TITLE
fix(cb2-7932): change state systems to run only on first change

### PR DESCRIPTION
**Pull request by naathanbrown**
## Ticket title

This adds on changes and makes it only run on inital load. The store now only pipes and changes the editing record when it is not falsy.
[https://dvsa.atlassian.net/browse/CB2-7932](7932)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

**Review by andrewf-bjss — DISMISSED** [review]

**Review by andrewf-bjss — APPROVED** [review]

File: src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.ts [hunk 1]
```diff
@@ -73,33 +75,37 @@ export class TechRecordSummaryComponent implements OnInit, OnDestroy {
...
-              this.techRecord.dimensions = { ...this.techRecord.dimensions, axleSpacing };
-              this.techRecord.axles = axles;
-            }
-
-            this.technicalRecordService.updateEditingTechRecord(this.techRecord, true);
-
-            return { ...cloneDeep(this.techRecord), reasonForCreation: '' };
-          }
+        skipWhile(editableTechRecord => !editableTechRecord),
+        map(editingTechRecord => {
```
Comment (tahayazici) [inline src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.ts:79]: Now that this is simplified, you can omit this `map()` altogether do this below:
```typescript
          this.techRecordCalculated = cloneDeep(techRecord);
```

File: src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.ts [hunk 2]
```diff
@@ -73,33 +75,37 @@ export class TechRecordSummaryComponent implements OnInit, OnDestroy {
...
+          this.sectionTemplates = this.vehicleTemplates;
+          this.middleIndex = Math.floor(this.sectionTemplates.length / 2);
+        }
       });
   }
 
+  ngOnChanges(changes: SimpleChanges): void {
+    const { techRecord } = changes;
+
+    if (techRecord.previousValue !== techRecord.currentValue && techRecord.isFirstChange()) {
```
Comment (tahayazici) [inline src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.ts:97]: Very clever way of using .isFirstChange(), I applaud you sir 👏 I presume this is tested and it works.

**Review by tahayazici — COMMENTED** [review]
Great work mate, just one comment to omit the `map()` which isn't necessary anymore.

File: src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.ts [hunk 3]
```diff
@@ -73,33 +75,37 @@ export class TechRecordSummaryComponent implements OnInit, OnDestroy {
...
-            }
-
-            this.technicalRecordService.updateEditingTechRecord(this.techRecord, true);
-
-            return { ...cloneDeep(this.techRecord), reasonForCreation: '' };
-          }
+        skipWhile(editableTechRecord => !editableTechRecord),
+        map(editingTechRecord => {
+          return cloneDeep(editingTechRecord);
         }),
```
Comment (gjulien-bjss) [inline src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.ts:81]:
```suggestion
        map(editingTechRecord => cloneDeep(editingTechRecord)),
```

**Review by andrewf-bjss — APPROVED** [review]

**Review by gjulien-bjss — APPROVED** [review]

**Review by shivangidas — APPROVED** [review]

**Pull request merged**